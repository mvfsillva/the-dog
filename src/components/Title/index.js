import React from 'react'
import PropTypes from 'prop-types'

const Title = ({ size: Size, color, ...props }) => <Size {...props} />

Title.propTypes = {
  size: PropTypes.string.isRequired,
  color: PropTypes.string,
}

Title.defaultProps = {
  size: 'h1',
  color: 'grayscale',
}

export default Title
