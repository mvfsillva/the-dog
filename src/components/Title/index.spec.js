import React from 'react'
import { shallow } from 'enzyme'
import Title from '.'

describe('[Component: Title]', () => {
  it('renders title component', () => {
    const text = 'Welcome the dog world'
    const size = 'h1'
    const title = shallow(<Title size={size}>{text}</Title>)
    expect(title.find(size).text()).toBe(text)
  })

  it('matches snapshot', () => {
    const wrapper = shallow(<Title>Title component</Title>)
    expect(wrapper).toMatchSnapshot()
  })
})
