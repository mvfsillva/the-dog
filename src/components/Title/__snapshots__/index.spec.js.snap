// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component: Title] matches snapshot 1`] = `
<h1>
  Title component
</h1>
`;
